File: Week-1_Design Patterns and Principles/2_FactoryMethodPatternExample/Code/ExcelDocument.cs
public class ExcelDocument : Document
{
    public ExcelDocument(string name) : base(name) {}
    public override void Open()
    {
        Console.WriteLine("Launching Excel sheet: " + name);
    }
    public override void Save()
    {
        Console.WriteLine("Saving spreadsheet " + name);
    }
}

File: Week-1_Design Patterns and Principles/2_FactoryMethodPatternExample/Code/ExcelDocument.java
class ExcelDocument extends Document {
    public ExcelDocument(String name) {
        super(name);
    }
    @Override
    void open() {
        System.out.println("Launching Excel sheet: " + name);
    }
    @Override
    void save() {
        System.out.println("Saving spreadsheet " + name);
    }
}